src/common/graph/sieve.js

```diff
@@ -93,34 +95,38 @@ function updateNode(current, index, change, result) {
   if (isBranch(change) && isBranch(node)) {
     // Both are branches: Recursively merge children.
     const nextResult = [];
+    node.version = change.version;
```

Comment:
This fixes a pre-existing bug where versions on the "branch" nodes were not updated even when versions on leaves were. We want to ensure that the subscription's internal state has the latest version number, even if the payload that bumped the version was suppressed.

Reply:
This required updates to several tests (see below) which were asserting the previous, incorrect, behavior.